contracts/keystore/L1/interfaces/IKeyStore.sol

```diff
@@ -0,0 +1,234 @@
+    #         where each service provider may have inconsistent response times.                                                           #
+    #         Therefore, providing the option for each user to set their own independent guardianSafePeriod is necessary.                 #
+    #                                                                                                                                     #
+    #######################################################################################################################################
+    */
+
+    /**
+     * @dev Emitted when the slot is initialized.
+     */
+    event Initialized(bytes32 indexed slot);
+
+    /**
+     * @dev Emitted when `key` is changed in `slot`.
+     */
+    event KeyChanged(bytes32 indexed slot, bytes32 key);
+
+    /**
+     * @dev Emitted when PreSetGuardian is called.
+     */
+    event SetGuardian(bytes32 indexed slot, bytes32 guardianHash, uint64 effectAt);
+
+    /**
+     * @dev Emitted when `guardianHash` is changed in `slot`.
+     */
+    event GuardianChanged(bytes32 indexed slot, bytes32 guardianHash);
+
+    /**
+     * @dev Emitted when CancelSetGuardian is called.
+     */
+    event CancelSetGuardian(bytes32 indexed slot, bytes32 guardianHash);
+
+    /**
+     * @dev Emitted when PreSetGuardianSafePeriod is called.
+     */
+    event SetGuardianSafePeriod(bytes32 indexed slot, uint64 guardianSafePeriod, uint64 effectAt);
+
+    /**
+     * @dev Emitted when `guardianSafePeriod` is changed in `slot`.
+     */
+    event GuardianSafePeriodChanged(bytes32 indexed slot, uint64 guardianSafePeriod);
+
+    /**
+     * @dev Emitted when CancelSetGuardianSafePeriod is called.
+     */
+    event CancelSetGuardianSafePeriod(bytes32 indexed slot, uint64 guardianSafePeriod);
+
+    /*
+        # Storage Layout
+
+        slot offset
+        ┌──────────┬────────────────────────────────┬────────────────┐
+        │  offset 0│ Key (EOA)                      │                │
+        ├──────────┼────────────────────────────────┤                │
+        │  offset 1│ nonce                          │                │
+        ├──────────┼────────────────────────────────┤                ├──────────────────┐
+        │  offset 2│ guardianHash                   │                │                  │
+        ├──────────┼────────────────────────────────┤                │                  │
+        │  offset 3│ pendingGuardianHash            │                │                  │
+        ├──────────┼────────────────────────────────┤  KeyStoreInfo  │                  │
+        │  offset 4│ guardianActivateAt             │                │                  │
+        ├──────────┼────────────────────────────────┤                │                  │
+        │  offset 5│ guardianSafePeriod             │                │   guardianInfo   │
```

Comment:
guardianSafePeriod type is defined uint64 below,  should it be the same offset with guardianActivateAt?